src/schema/nmdc.yaml

```diff
@@ -242,15 +242,28 @@ classes:
   FunctionalAnnotationAggMember:
     class_uri: nmdc:FunctionalAnnotationAggMember
     slots:
-      - metagenome_annotation_id
+      - was_generated_by
       - gene_function_id
       - count
       - type
     slot_usage:
-      metagenome_annotation_id:
+      was_generated_by:
         structured_pattern:
-          syntax: "{id_nmdc_prefix}:wfmgan-{id_shoulder}-{id_blade}{id_version}{id_locus}$"
+          syntax: "{id_nmdc_prefix}:(wfmgan|wfmp)-{id_shoulder}-{id_blade}{id_version}$"
           interpolated: true
+        required: true
+    rules:
```

Comment:
no value in storing counts for metap?  Any chance this info would be useful for future visualizations?

Reply:
I'll double check with protein people, right now we are not planning on having count information in the metadata (see https://github.com/microbiomedata/berkeley-schema-fy24/pull/236), but I'll double check that is what we want thinking about future visualizations.

Reply:
Of course, it would be good to know what we are counting now as part of the aggregation...?

Things that 'can' be counted include number of MS2 spectra assigned to a GeneProduct for a given WorkflowExecution, the number of peptide sequences observed for a given GeneProduct, or number of (razor_)protein entries slash GeneProducts assigned to a functional annotation within a given WorkflowExecution, probably moar. 

For visualization I 'think' the spectral counts per GeneProduct is most relevant (basically a form of quantitation called, surprisingly enough, Spectral Counting), especially if normalized to the highest count number across all of the GeneProducts found in the WorkflowExecution.

This would require more information being brought in, adding some bloat to the de-bloat. I would expect we could reasonably easily add this information after the re-imagining of this aggregation versus trying to design this and re-re-re-design the de-bloat effort's output, but maybe now is the time since we're under the hood. Of course, I'm also not the Decider for this...

Reply:
I'm inclined to agree with you @SamuelPurvine re: spectral counts per GeneProduct.  Is it reasonable to provide the # of MS2 spectra mapping to each `razor_protein`?  If so, I can revisit the modeling on `MetaproteomicsAnalysis` (#236) and think of how we'd keep that info and migrate it from our current data - it's not too bad I don't think.

I think the idea is that the count slot on this aggregation could be flexible to match the data type its aggregating (and I plan on adding a description to that slot so we'll know!).

Reply:
Thanks @aclum for helping us keep perspective on what we might want in the future.

Reply:
Removing count rule, we've decided to keep some count data for proteomics.  I'll add to the description that this is spectral counts if proteomics data, sequence counts for MetaG/MetaT.